File: app/src/main/java/chenjian/aplay/MainActivity.java
package chenjian.aplay;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;

import android.Manifest;
import android.app.Activity;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    // Used to load the 'native-lib' library on application startup.
    static {
        System.loadLibrary("native-lib");
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        PermissionUtil.verifyStoragePermissions(this);

        // Example of a call to a native method
        //TextView tv = findViewById(R.id.sample_text);
        //open("/sdcard/DCIM/Camera/test1.mp4", this);
        //tv.setText(stringFromJNI());
        //startActivity(new Intent(this.getApplicationContext(), TestActivity.class));
    }

    /**
     * A native method that is implemented by the 'native-lib' native library,
     * which is packaged with this application.
     */
    public native String stringFromJNI();

    public native boolean open(String url, Object handle);


}

File: app/src/main/java/chenjian/aplay/MainActivity.kt
package chenjian.aplay

import android.os.Bundle
import android.os.Handler
import android.util.Log
import android.view.WindowManager
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import chenjian.aplay.palyview.PlayListener

import chenjian.aplay.palyview.VideoView

class MainActivity : AppCompatActivity() {
    private var mVideoView: VideoView? = null
    val handler : Handler = Handler()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        PermissionUtil.verifyStoragePermissions(this)
        mVideoView = findViewById(R.id.video_view)

//        handler.postDelayed(Runnable {
//            var layoutParam = mVideoView!!.layoutParams
//            layoutParam.height = WindowManager.LayoutParams.MATCH_PARENT
//            layoutParam.width = WindowManager.LayoutParams.MATCH_PARENT;
//            mVideoView!!.layoutParams = layoutParam
//        }, 5000)
    }

    override fun onResume() {
        super.onResume()
        // TODO: 2020/9/1 路径需要可以选择
        mVideoView!!.setDataSource("/sdcard/DCIM/Camera/111.mp4")
        mVideoView!!.addPlayListener(object : PlayListener {
            override fun onPrepare() {
                Log.i("testFFmpeg", "******* java onPrepare() *******")
                mVideoView!!.start()
            }

            override fun onProgress(progress: Int) {
                Log.i("testFFmpeg", "******* java onProgress()  ******* $progress")
            }

            override fun onError(errorCode: Int) {
                Log.i("testFFmpeg", "******* java onError() ******* $errorCode")
            }
        })

        mVideoView!!.addPlayListener(object : PlayListener {
            override fun onPrepare() {
                Log.i("testFFmpeg", "******* 22222 java onPrepare() *******")
                mVideoView!!.start()
            }

            override fun onProgress(progress: Int) {
                Log.i("testFFmpeg", "******* 22222 java onProgress()  ******* $progress")
            }

            override fun onError(errorCode: Int) {
                Log.i("testFFmpeg", "******* 2222 java onError() ******* $errorCode")
            }
        })
        mVideoView!!.prepare()
    }
}